lib/plugins/aws/package/compile/events/s3/index.js

```diff
@@ -74,21 +102,32 @@ class AwsCompileS3Events {
                 filter = { Filter: { S3Key: { Rules: rules } } };
               }
             } else if (typeof event.s3 === 'string') {
-              bucketName = event.s3;
+              bucket = event.s3;
             } else {
               const errorMessage = [
                 `S3 event of function ${functionName} is not an object nor a string.`,
-                ' The correct syntax is: s3: bucketName OR an object with "bucket" property.',
+                ' The correct syntax is: s3: bucketName/provider bucket id or an object with "bucket" property.',
                 ' Please check the docs for more info.',
               ].join('');
               throw new this.serverless.classes.Error(errorMessage);
             }
 
             const lambdaLogicalId = this.provider.naming.getLambdaLogicalId(functionName);
+            let bucketName;
+            if (provderS3[bucket]) {
+              bucketName = provderS3[bucket].name || bucket;
+              logicalId = this.provider.naming.getBucketLogicalId(bucket);
+            } else {
+              bucketName = bucket;
+            }
+
+            if (logicalId) {
+              bucketsLogicalIds[bucketName] = { logicalId, bucket };
```

Comment:
Let's put _name_ as `name` (not `bucket`)

Reply:
Thats not always the name

Reply:
So if it's our reference name, let's name it `ref`